April news (#1287)
ropensci/roweb3: content/blog/2026-04-30-news-april-2026/index.Rmd

english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(trimws(...), sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## Tomáš Kalibera (1978–2026)

The rOpenSci team is deeply saddened at the loss of Tomáš Kalibera, a member of R-Core and contributor to the R community for almost 10 years, who passed away on April 1.  
Our thoughts are with Tomáš's friends and family at this time.  

Read [Jan Vitek's remembrance of Tomáš](https://prl-prg.github.io/tomas-kalibera.html).

## rOpenSci HQ

### New editors Alec Robitaille and Lucy D’Agostino McGowan

We’re excited to welcome Alec Robitaille and Lucy D’Agostino McGowan as new editors. 
Alec joins our general review team, and Lucy our statistical software review team. 
Read more in the [post introducing them](/blog/2026/04/16/editors2026/)!

### Champions Program Update

> TODO YANI?

Reviewer: ```suggestion
We’re excited to share that we’ve finished selecting the new cohort for the rOpenSci Champions Program!
This was not an easy process. The quality of the proposals was exceptionally high, which made the selection both challenging and inspiring. We’re grateful to everyone who applied and shared their ideas with us.
Please join us in welcoming our new Champions: Bastián Olea Herrera, Durga Valentina Linares Herrera, José Daniel Conejeros, Denisse Fierro Arcos, Evelia Lorena Coss Navarrete, Gladys Choque Ulloa, Linda Jazmín Cabrera Orellana, Patricia Andrea Loto, Marina Cecilia Cock, María Florencia Tames, and Estefanía Torrejón. 

Over the coming months, they will contribute to the R Community through developing new packages, reviewing, and submitting packages for peer review.
We’re looking forward to working with this amazing group and supporting their projects!
```

---
slug: "news-april-2026"
title: rOpenSci News Digest, April 2026
author:
  - The rOpenSci Team
date: '2026-04-30'
tags:
  - newsletter

> description: Bioconductor collaboration, R/Medecine, goodpractice, targets VSCode extension, new packages and package news

Reviewer 2: ```suggestion
description: Bioconductor collaboration, R/Medicine, goodpractice, targets VSCode extension, new packages and package news
```

params:
  last_newsletter: '2026-03-30'
  doi: "10.59350/4qpnx-96e10"
---

```{r setup, include=FALSE}
library("rlang")
last_newsletter <- anytime::anytime(params$last_newsletter)
knitr::opts_chunk$set(echo = FALSE)
url <- sprintf(
    "/blog/%s/%s/%s/%s",
    lubridate::year(rmarkdown::yaml_front_matter(knitr::current_input())$date),
    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(trimws(...), sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## Tomáš Kalibera (1978–2026)

> The rOpenSci team is deeply saddened at the loss of Tomáš Kalibera, a member of R-Core and contributor to the R community for almost 10 years, who passed away on April 1.  

Reviewer 2: ```suggestion
The rOpenSci team is deeply saddened at the loss of Tomáš Kalibera, a member of R-Core and contributor to the R community for almost 10 years. Tomáš passed away on April 1.  
```

Our thoughts are with Tomáš's friends and family at this time.  

Read [Jan Vitek's remembrance of Tomáš](https://prl-prg.github.io/tomas-kalibera.html).

## rOpenSci HQ

### New editors Alec Robitaille and Lucy D’Agostino McGowan

> We’re excited to welcome Alec Robitaille and Lucy D’Agostino McGowan as new editors. 

Reviewer 2: ```suggestion
We’re excited to welcome [Alec Robitaille](/author/alec-robitaille/) and [Lucy D’Agostino McGowan](/author/lucy-dagostino-mcgowan/) as new editors. 
```

Alec joins our general review team, and Lucy our statistical software review team. 
Read more in the [post introducing them](/blog/2026/04/16/editors2026/)!

> ### Champions Program Update

Reviewer 2: ```suggestion
### Champions Program update
```

We’re excited to share that we’ve finished selecting the new cohort for the rOpenSci Champions Program!
This was not an easy process. The quality of the proposals was exceptionally high, which made the selection both challenging and inspiring. We’re grateful to everyone who applied and shared their ideas with us.
Please join us in welcoming our new Champions: Bastián Olea Herrera, Durga Valentina Linares Herrera, José Daniel Conejeros, Denisse Fierro Arcos, Evelia Lorena Coss Navarrete, Gladys Choque Ulloa, Linda Jazmín Cabrera Orellana, Patricia Andrea Loto, Marina Cecilia Cock, María Florencia Tames, and Estefanía Torrejón. 

> Over the coming months, they will contribute to the R Community through developing new packages, reviewing, and submitting packages for peer review.

Reviewer 2: ```suggestion
Over the coming months, they will contribute to the R Community through developing new packages, reviewing packages, and submitting packages for peer review.
```

We’re looking forward to working with this amazing group and supporting their projects!

### Collaborating between Bioconductor and R-universe on Development of Common Infrastructure

Bioconductor is collaborating with R-universe to gradually modernize parts of its infrastructure, while accommodating the project’s scale, governance, and established processes. 
In turn, Bioconductor is helping R-universe expand and refine its features as we learn to serve the complex needs of the Bioconductor community.
Read more in the [blog post](/blog/2026/04/08/r-universe-bioc/).

### Yanina Bellini Saibene at R/Medicine 2026

Yanina Bellini Saibene will deliver her keynote talk ["Software Sustainability and Community Management"](/events/r-medicine-2026-keynote/) on Thursday May 7th, 11:15AM–12:15PM ET.

### Jeroen Ooms at "Where Do R Packages Live?"

Jeroen Ooms will take part in an online discussion panel on Wednesday 20 May at 5:00 PM - 6:00 PM (AEST).
The panel is organized by the Statistical Computing and Visualisation section of the Statistical Society of Australia.

> [More details](https://statsoc.org.au/event-6653060).

Reviewer 2: ```suggestion
**Jeroen Ooms at "Where Do R Packages Live?"**
Jeroen will take part in an online discussion panel on Wednesday 20 May at 5:00 PM - 6:00 PM (AEST).
The panel is organized by the Statistical Computing and Visualisation section of the Statistical Society of Australia ([details](https://statsoc.org.au/event-6653060)).
```

> [More details](https://statsoc.org.au/event-6653060).

Reviewer 2: Perhaps consolidate and simplify a bit, these two items?
```suggestion
### rOpenSci Staff presentations

#### Yanina Bellini Saibene at R/Medicine 2026

Yani will deliver her keynote talk ["Software Sustainability and Community Management"](/events/r-medicine-2026-keynote/) on Thursday May 7th, 11:15AM–12:15PM ET.

#### Jeroen Ooms at "Where Do R Packages Live?"

Jeroen will take part in an online discussion panel on Wednesday 20 May at 5:00 PM - 6:00 PM (AEST).
The panel is organized by the Statistical Computing and Visualisation section of the Statistical Society of Australia ([details](https://statsoc.org.au/event-6653060)).
```

### Updates to the goodpractice R package

We have long recommended the [goodpractice package](https://docs.ropensci.org/goodpractice), which identifies issues with R packages, and advises how to fix them.
Thanks to a huge amount of work by valued community member [Athanasia Mo Mowinckel](/author/athanasia-mo-mowinckel/), goodpractice has been extended and improved to include entirely new suites of checks, and improved ability to control which checks are run.

> A blog post describing the updates will be published very soon, but in the meantime, we encourage you to install the current [development version](https://docs.ropensci.org/goodpractice/#installation) and try it out yourself.

Reviewer 2: The 'very' sounds almost too eager, perhaps a bit apologetic? No need for that 😉 Also, it might get delayed given recent schedule changes.

```suggestion
A blog post describing the updates will be published soon, but in the meantime, we encourage you to install the current [development version](https://docs.ropensci.org/goodpractice/#installation) and try it out yourself.
```

### Analyse your targets pipeline

Following our [blog post about tree-sitter](/blog/2026/04/02/tree-sitter-overview/), Tyler Morgan Wall was [inspired](https://news.ycombinator.com/item?id=47801899) to create a [static analysis tool for targets pipelines (as a VS Code extension)](https://github.com/tylermorganwall/tarborist).

### Coworking 

Read [all about coworking](/blog/2023/06/21/coworking/)!

- Tuesday May 5th 2026, 9:00 Australia Western (01:00 UTC) ["Code Review with rOpenSci"](/events/coworking-2026-05/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Liz Hare](/author/liz-hare/).  
    -   Explore resources for Code Review
    -   Sign up to volunteer to do [software peer-review](https://airtable.com/app8dssb6a7PG6Vwj/shrnfDI2S9uuyxtDw) at rOpenSci
    -   Meet cohost, Liz Hare, and discuss resources for Code Review with rOpenSci.

> 

Reviewer 2: @maelle, I still need to see if I will have something for June, and I will merge Mark's event too. I will update this by Wednesday end of your day. Okay?

Author: Yes!

> And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

Reviewer 2: Placeholder as I don't think I'm likely to get this sorted by tomorrow!
```suggestion
    -   Meet cohost, Liz Hare, and discuss resources for Code Review with rOpenSci.
 
 - Tuesday June 2nd 2026, 14:00 Europe Central (12:00 UTC) [theme to be determined], with [Steffi LaZerte](/author/steffi-lazerte/) and cohost to be determined.  
    -   Explore resources related to the theme
    -   Meet the cohost, and other attendees, and discuss the theme or other topics.

And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!
```

> 

Reviewer 2: @maelle, what do you think of this format for the Office hours?

```suggestion
And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

### Editors' Office Hours

We are exploring hosting a new event, Editors' Office Hours, where you can drop in to ask questions about rOpenSci Software Peer Review on or near the third Tuesday of each month, alternating among timezones to accommodate different parts of the world.

Upcoming office hours:

- Tuesday May 19, 16:00-17:00 Europe Central (14:00-15:00 UTC) ((event)[/events/office-hours-2026-05/))
```

Author: great!

> ## Software :package:

Reviewer: ```suggestion
### useR! 2026 Diversity Scholarship Program

useR! 2026 is offering diversity scholarships to support participation from people in underrepresented or historically marginalized groups within the R community. The program includes both registration fee waivers and full needs-based scholarships, which cover conference registration as well as travel and lodging (via reimbursement). Applications are open to eligible participants worldwide **until May 10**, and will be reviewed based on need, eligibility, and potential impact by a committee from Forwards, RLadies+ Global, and rOpenSci.

Found all the details and important links on the conference website: https://user2026.r-project.org/additional/diversity_scholarship.html

## Software :package:
```

## Calls for contributions

### Calls for maintainers

If you're interested in maintaining any of the R packages below, you might enjoy reading our blog post [What Does It Mean to Maintain a Package?](/blog/2023/02/07/what-does-it-mean-to-maintain-a-package/).

- [landscapetools](https://docs.ropensci.org/landscapetools), R package for some of the less-glamorous tasks involved in landscape analysis. [Issue for volunteering](https://github.com/ropensci/landscapetools/issues/48).

- [hddtools](https://docs.ropensci.org/hddtools), Tools to discover hydrological data, accessing catalogues and databases from various data providers. [Issue for volunteering](https://github.com/ropensci/hddtools/issues/36).
        
- [qualtRics](https://docs.ropensci.org/qualtRics/), download Qualtrics survey data. [Issue for volunteering](https://github.com/ropensci/qualtRics/issues/383).

### Calls for contributions

Refer to our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

## Package development corner

Some useful information for R package developers. :eyes:

### Useless code, duplicated function? Refactoring with Jarl

> The Jarl CLI by Étienne Bacher got useful new features for package developers:

Reviewer 2: ```suggestion
The Jarl CLI by Étienne Bacher received several useful new features for package developers:
```

- [`unused_function`](https://jarl.etiennebacher.com/rules/unused_function) 
- [`duplicated_function_definition`](https://jarl.etiennebacher.com/rules/duplicated_function_definition).

They are a nice complement to [`unreachable_code`](https://jarl.etiennebacher.com/rules/unreachable_code).

Read more in the [release announcement](https://www.etiennebacher.com/posts/2026-03-23-jarl-0.5.0/#other-features-for-package-developers).

### Git commands to get to know a project

Ally Piechowski wrote an insightful post entitled ["The Git Commands I Run Before Reading Any Code"](https://piechowski.io/post/git-commands-before-reading-code/),

> for instance a command to determine which files recently changed the most!

Reviewer 2: ```suggestion
Ally Piechowski wrote an insightful post entitled ["The Git Commands I Run Before Reading Any Code"](https://piechowski.io/post/git-commands-before-reading-code/), which suggests Git commands which are useful for understanding the code base. For instance a command to determine which files recently changed the most!
```

> Garrick Aden-Buie wrote ["a little `git-recon` bash script that runs them in series, complete with some ascii bar plots."](https://gist.github.com/gadenbuie/463ff1e9f3b0f48cddc44db2224d286b).

Reviewer 2: ```suggestion
To complement this post, Garrick Aden-Buie wrote ["a little `git-recon` bash script that runs them in series, complete with some ascii bar plots"](https://gist.github.com/gadenbuie/463ff1e9f3b0f48cddc44db2224d286b).
```

### Another R mailing list archive

Last month we listed several ways to [browse the R mailing lists](/blog/2026/03/30/news-mars-2026/#how-to-browse-the-r-mailing-lists).
Newsletter reader Florian Kohrt kindly wrote to us to mention the [plain text backup](https://github.com/MichaelChirico/r-mailing-list-archive) maintained by Michael Chirico.

### Will R run out of random seeds? Useful seed explainer

Andrew Heiss published an useful and interesting deep dive into [random seeds](https://www.andrewheiss.com/blog/2026/04/13/seeds-predetermined-universes/).

### Enforcing the coalesce operator

> If you enjoy the coalesce operator `%||%` introduced in [R 4.4.0](https://cran.r-project.org/bin/windows/base/old/4.4.0/NEWS.R-4.4.0.html)

Reviewer 2: ```suggestion
Are you enjoying the coalesce operator `%||%` introduced in [R 4.4.0](https://cran.r-project.org/bin/windows/base/old/4.4.0/NEWS.R-4.4.0.html)?
```

> `L %||% R` newly in base is an expressive idiom for the phrases `if(!is.null(L)) L else R or if(is.null(L)) R else L`. 

> You can enforce it via [Jarl](https://jarl.etiennebacher.com/rules/coalesce) or [lintr](https://lintr.r-lib.org/reference/coalesce_linter.html).

Reviewer 2: ```suggestion
Consider enforcing it via [Jarl](https://jarl.etiennebacher.com/rules/coalesce) or [lintr](https://lintr.r-lib.org/reference/coalesce_linter.html).
```

> As a reminder, the operator is supported by the [backports](https://github.com/r-lib/backports/pull/81/changes) R package and is exported by rlang.

Reviewer 2: Is it "also" exported by rlang? Or is that related to the backports?
```suggestion
As a reminder, the operator can be used in older versions of R through the [backports](https://github.com/r-lib/backports/pull/81/changes) R package or by importing it from [rlang](https://rlang.r-lib.org/reference/op-null-default.html).
```

> ### Positive AI reading

Reviewer 2: Lovely!
